### PR TITLE
Added an optional Unit type to IngredientQuantityByProcessAndName

### PR DESCRIPTION
**Pull request by dleblanc**
This adds the optional 'unit' to IngredientQuantityByProcessAndName, necessary for a proper filter using absolute quantities.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [x] I have bumped the version in setup.py

### CODE REVIEW

**Comment by dleblanc** [comment]
Looks like this is necessary to get in so the devkit e2e's can supply a now-mandatory unit to the backend when searching for absolute quantities.

File: src/citrine/gemtables/variables.py [hunk 1]
```diff
@@ -421,13 +422,15 @@ def __init__(self, *,
                  process_template: LinkByUID,
                  ingredient_name: str,
                  quantity_dimension: IngredientQuantityDimension,
-                 type_selector: DataObjectTypeSelector = DataObjectTypeSelector.PREFER_RUN):
+                 type_selector: DataObjectTypeSelector = DataObjectTypeSelector.PREFER_RUN,
+                 unit: str = None):
```
Comment (maxhutch) [inline src/citrine/gemtables/variables.py:426]:
```suggestion
                 unit: Optional[str] = None):
```

**Review by maxhutch — CHANGES_REQUESTED** [review]
`unit` should be added to the docstring, with a description of when it needs to be set.  We _could_ even raise a ValueError here in citrine-python if the unit is `None` and the quantity_dimension is ABSOLUTE.

I do not think there is anywhere in the user guide that needs to be update.

I think we should be adding this option to `IngredientQuantityInOutput` too (cc @kroenlein ).

It would be good to set `unit` in one of the unit test cases, probably [this one](https://github.com/CitrineInformatics/citrine-python/blob/master/tests/gemtable/test_variables.py#L20)

File: setup.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ def run(self):
 
 
 setup(name='citrine',
-      version='0.71.0',
+      version='0.71.1',
```
Comment (maxhutch) [inline setup.py:60]: I think a new option should be a minor version bump.

Comment (dleblanc) [reply setup.py:60]: Yeah good point.

File: src/citrine/gemtables/variables.py (hunk 1 above)
Comment (dleblanc) [reply src/citrine/gemtables/variables.py:426]: Updated the docstring, and added the unit to the test case.

With regards to adding the unit to IngredientQuantityInOutput, maybe that should be a separate story, as this is focused on unblocking default table generation.

**Review by kroenlein — COMMENTED** [review]
It seems a bit inconsistent to implement for IngredientQuantityByProcessAndName but not IngredientQuantityInOutput.

As per our conversation in stand-up, we're carding this up as a second ticket to prioritize, since there's no obvious synergy between the implementations.

https://citrine.atlassian.net/browse/PLA-5736

**Review by maxhutch — APPROVED** [review]

**Pull request merged**